client/web/src/components/LinkWithIcon.tsx

```diff
@@ -1,24 +1,43 @@
 import React from 'react'
-import { Link, NavLink } from 'react-router-dom'
-import { kebabCase } from 'lodash'
+import classNames from 'classnames'
+import { kebabCase, omit } from 'lodash'
+import { NavLink, NavLinkProps } from 'react-router-dom'
+import { LinkProps, Link } from '../../../shared/src/components/Link'
+
+interface Props extends LinkProps, Pick<NavLinkProps, 'activeClassName'> {
+    text: string
+}
+
+interface PropsWithIcon extends Props {
+    icon: React.ComponentType<{ className?: string }>
+}
+
+interface PropsWithIconPlaceholder extends Props {
+    hasIconPlaceholder: true
+}
 
 /**
  * A link displaying an icon along with text.
  *
  */
-export const LinkWithIcon: React.FunctionComponent<{
-    to: string
-    text: string
-    icon: React.ComponentType<{ className?: string }>
-    className?: string
-    activeClassName?: string
-}> = ({ to, text, icon: Icon, className = '', activeClassName }) => {
+export const LinkWithIcon: React.FunctionComponent<PropsWithIcon | PropsWithIconPlaceholder> = props => {
+    const { to, text, className = '', activeClassName, ...restProps } = props
     const LinkComponent = activeClassName ? NavLink : Link
-    const linkProps = { to, className: `${className} d-flex align-items-center`, activeClassName }
+
+    // use `svg` element as a placeholder when `hasIconPlaceholder` is true
+    const Icon = 'hasIconPlaceholder' in props ? 'svg' : props.icon
+
+    const linkProps = {
+        to,
+        className: classNames('d-flex', 'align-items-center', className),
+        ...(activeClassName && { activeClassName }),
```

Comment:
```suggestion
        activeClassName,
```

This should be the same I believe

Reply:
React will complain about unknown HTML attribute: `React does not recognize the `activeClassName` prop on a DOM element`